packages/multichain/src/caip25Permission.ts

```diff
@@ -0,0 +1,283 @@
+      const existingEvmAddresses = methodHooks
+        .listAccounts()
+        .map((account) => account.address);
+      const ethAccounts = getEthAccounts({
+        requiredScopes: normalizedRequiredScopes,
+        optionalScopes: normalizedOptionalScopes,
+      }).map((address) => address.toLowerCase() as Hex);
+
+      const allEthAccountsSupported = ethAccounts.every((address) =>
+        existingEvmAddresses.includes(address),
+      );
+      if (!allEthAccountsSupported) {
+        throw new Error(
+          `${Caip25EndowmentPermissionName} error: Received eip155 account value(s) for caveat of type "${Caip25CaveatType}" that were not found in the wallet keyring.`,
+        );
+      }
+
+      if (
+        !isEqual(requiredScopes, normalizedRequiredScopes) ||
+        !isEqual(optionalScopes, normalizedOptionalScopes)
+      ) {
+        throw new Error(
+          `${Caip25EndowmentPermissionName} error: Received non-normalized value for caveat of type "${Caip25CaveatType}".`,
+        );
+      }
+    },
+  };
+};
+
+/**
+ * The `caip25` endowment specification builder. Passed to the
+ * `PermissionController` for constructing and validating the
+ * `endowment:caip25` permission.
+ */
+export const caip25EndowmentBuilder = Object.freeze({
+  targetName: Caip25EndowmentPermissionName,
+  specificationBuilder,
+} as const);
+
+/**
+ * Factories that construct caveat mutator functions that are passed to
+ * PermissionController.updatePermissionsByCaveat.
+ */
+export const Caip25CaveatMutators = {
+  [Caip25CaveatType]: {
+    removeScope,
+    removeAccount,
+  },
+};
+
+/**
+ * Removes the account from the scope object.
+ *
+ * @param targetAddress - The address to remove from the scope object.
+ * @returns A function that removes the account from the scope object.
+ */
+function removeAccountFilterFn(targetAddress: string) {
+  return (account: CaipAccountId) => {
+    const parsed = parseCaipAccountId(account);
+    return parsed.address !== targetAddress;
+  };
+}
+
+/**
+ * Removes the account from the scope object.
+ *
+ * @param scopeObject - The scope object to remove the account from.
+ * @param targetAddress - The address to remove from the scope object.
+ */
+function removeAccountFromScopeObject(
+  scopeObject: InternalScopeObject,
+  targetAddress: string,
+) {
+  if (scopeObject.accounts) {
+    scopeObject.accounts = scopeObject.accounts.filter(
+      removeAccountFilterFn(targetAddress),
+    );
+  }
+}
+
+/**
+ * Removes the target account from the scope object.
+ *
+ * @param caip25CaveatValue - The CAIP-25 permission caveat value from which to remove the account (across all chain scopes).
+ * @param targetAddress - The address to remove from the scope object. Not a CAIP-10 formatted address because it will be removed across each chain scope.
+ * @returns The updated scope object.
+ */
+function removeAccount(
+  caip25CaveatValue: Caip25CaveatValue,
+  targetAddress: string,
```

Comment:
Can we specify that this is a Hex address here? It seems that we are assuming this is the format.
```suggestion
  targetAddress: Hex,
```